### PR TITLE
Bc4710 Integrate tldraw board in the column

### PR DESCRIPTION
**Pull request by VikDavydiuk**
# Description

## Links to Tickets or other pull requests

- hpi-schul-cloud/nuxt-client#2839
- hpi-schul-cloud/dof_app_deploy#618
- hpi-schul-cloud/schulcloud-server#4352
- https://ticketsystem.dbildungscloud.de/browse/BC-4710

## Approval for review

- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### CODE REVIEW

File: .github/workflows/clean.yml [hunk 1]
```diff
@@ -7,9 +7,9 @@ jobs:
   clean:
     permissions:
       contents: read
-    uses: hpi-schul-cloud/dof_app_deploy/.github/workflows/clean_workflow.yml@main
+    uses: hpi-schul-cloud/dof_app_deploy/.github/workflows/clean_workflow.yml@BC-4710-new-tldraw-manage
```
Comment (davwas) [inline .github/workflows/clean.yml:10]: This will need to be changed to main before merging, right?

File: .github/workflows/push.yml [hunk 2]
```diff
@@ -68,13 +86,14 @@ jobs:
     needs:
       - build_and_push
       - branch_name
-    uses: hpi-schul-cloud/dof_app_deploy/.github/workflows/deploy.yml@main
+    uses: hpi-schul-cloud/dof_app_deploy/.github/workflows/deploy.yml@BC-4710-new-tldraw-manage
```
Comment (davwas) [inline .github/workflows/push.yml:89]: This will need to be changed to main before merging, right?

**Review by davwas — APPROVED** [review]

File: ansible/roles/tldraw-client-core/templates/tldraw-client-deployment.yml.j2 [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  replicas: {{ TLDRAW_CLIENT_REPLICAS|default("1", true) }}
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: 1
+  revisionHistoryLimit: 4
+  paused: false
+  selector:
+    matchLabels:
+      app: tldraw-client
```
Comment (mamutmk5) [inline ansible/roles/tldraw-client-core/templates/tldraw-client-deployment.yml.j2:18]: We add no more labels after hpi-schul-cloud/schulcloud-server#4517 was merged

```suggestion
  labels:
    app: tldraw-client
    app.kubernetes.io/part-of: schulcloud-verbund
    app.kubernetes.io/version: {{ SCHULCLOUD_SERVER_IMAGE_TAG }}
    app.kubernetes.io/name: tldraw-client
    app.kubernetes.io/component: tldraw
    app.kubernetes.io/managed-by: ansible
    git.branch: {{ SCHULCLOUD_SERVER_BRANCH_NAME }}
    git.repo: {{ SCHULCLOUD_SERVER_REPO_NAME }}
```

File: ansible/roles/tldraw-client-core/templates/tldraw-client-deployment.yml.j2 [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  template:
+    metadata:
+      labels:
+        app: tldraw-client
```
Comment (mamutmk5) [inline ansible/roles/tldraw-client-core/templates/tldraw-client-deployment.yml.j2:22]: We add no more labels after hpi-schul-cloud/schulcloud-server#4517 was merged

```suggestion
      labels:
        app: tldraw-client
        app.kubernetes.io/part-of: schulcloud-verbund
        app.kubernetes.io/version: {{ SCHULCLOUD_SERVER_IMAGE_TAG }}
        app.kubernetes.io/name: tldraw-client
        app.kubernetes.io/component: tldraw
        app.kubernetes.io/managed-by: ansible
        git.branch: {{ SCHULCLOUD_SERVER_BRANCH_NAME }}
        git.repo: {{ SCHULCLOUD_SERVER_REPO_NAME }}
```

**Review by mamutmk5 — CHANGES_REQUESTED** [review]
Missing the igress definition for the tldwar-client at the ansible role for thldraw-client

**Review by mamutmk5 — CHANGES_REQUESTED** [review]
missing the push to `quay.io` for the taged tldraw-client docker image at `.github/workflows/tag.yml` like it's done at https://github.com/hpi-schul-cloud/schulcloud-server/blob/main/.github/workflows/tag.yml#L19C1-L49

File: ansible/roles/tldraw-client-core/templates/tldraw-client-ingress.yml.j2 [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  - host: {{ DOMAIN }}
+    http:
+      paths:
+      - path: /tldraw
+        pathType: Prefix
+        backend:
+          service:
+            name: tldraw-client-svc
+            port:
+              number: {{ PORT_TLDRAW_CLIENT }}
```
Comment (mamutmk5) [inline ansible/roles/tldraw-client-core/templates/tldraw-client-ingress.yml.j2:35]: code the port hard :)
```suggestion
              number: 3046
```

File: ansible/roles/tldraw-client-core/templates/tldraw-client-ingress.yml.j2 [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+      - path: /tldraw-client-runtime.config.json
+        pathType: Prefix
+        backend:
+          service:
+            name: tldraw-client-svc
+            port:
+              number: {{ PORT_TLDRAW_CLIENT }}
```
Comment (mamutmk5) [inline ansible/roles/tldraw-client-core/templates/tldraw-client-ingress.yml.j2:42]: this Ingress rule for the path `/tldraw-client-runtime.config.json` is not needed.
Why? We uns the pathType: Prefix
That mean if the ingress Rule for `/tldraw` matche an path is go to the tldraw-clinet-svc and in the end to the pod.
The path _**/tldraw**-client-runtime.config.json_ is includet in **/tldraw**

Other case where
```
      - path: /tldraw/
        pathType: Prefix
        backend:
          service:
            name: tldraw-client-svc
            port:
              number: {{ PORT_TLDRAW_CLIENT }}
```
Then the ingress rule woud only match all belwo the path `/tldraw/`

File: ansible/roles/tldraw-client-core/templates/tldraw-client-deployment.yml.j2 [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+        runAsGroup: 1000
+        fsGroup: 1000
+        runAsNonRoot: true
+      containers:
+      - name: tldraw-client
+        image: {{ TLDRAW_CLIENT_IMAGE }}:{{ TLDRAW_CLIENT_IMAGE_TAG }}
+        imagePullPolicy: IfNotPresent
+        ports:
+        - name: http
+          containerPort: {{ PORT_TLDRAW_CLIENT }}
```
Comment (mamutmk5) [inline ansible/roles/tldraw-client-core/templates/tldraw-client-deployment.yml.j2:36]:
```suggestion
          containerPort: 3046
```

File: ansible/roles/tldraw-client-core/templates/tldraw-client-ingress.yml.j2 [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+      - path: /static
+        pathType: Prefix
+        backend:
+          service:
+            name: tldraw-client-svc
+            port:
+              number: {{ PORT_TLDRAW_CLIENT }}
```
Comment (mamutmk5) [inline ansible/roles/tldraw-client-core/templates/tldraw-client-ingress.yml.j2:49]:
```suggestion
              number: 3046
```

File: ansible/group_vars/all/tldraw-client.yml [hunk 9]
```diff
@@ -0,0 +1,3 @@
+---
+PORT_TLDRAW_CLIENT: 3046
```
Comment (mamutmk5) [inline ansible/group_vars/all/tldraw-client.yml:2]: shoud be not needed after every place where `PORT_TLDRAW_CLIENT` is useds has the hard codeed value 3046

**Review by mamutmk5 — CHANGES_REQUESTED** [review]

File: ansible/roles/tldraw-client-core/templates/tldraw-client-ingress.yml.j2 [hunk 10]
```diff
@@ -0,0 +1,43 @@
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: {{ NAMESPACE }}-tldraw-client-ingress
+  namespace: {{ NAMESPACE }}
+  annotations:
+    nginx.ingress.kubernetes.io/ssl-redirect: "{{ TLS_ENABLED | default("false") }}"
+    nginx.ingress.kubernetes.io/auth-type: basic
+    nginx.ingress.kubernetes.io/auth-secret: shd-basic-auth-secret
```
Comment (mamutmk5) [inline ansible/roles/tldraw-client-core/templates/tldraw-client-ingress.yml.j2:9]: Remove this Line to remove the HTTP-Basic Auth request.
Please lock if you do copy&past from other things like Ingresses what you copy.

**Review by mamutmk5 — CHANGES_REQUESTED** [review]

File: ansible/roles/tldraw-client-core/templates/tldraw-client-ingress.yml.j2 [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+    nginx.ingress.kubernetes.io/auth-realm: 'Authentication Required'
```
Comment (mamutmk5) [inline ansible/roles/tldraw-client-core/templates/tldraw-client-ingress.yml.j2:10]: https://github.com/hpi-schul-cloud/schulcloud-server/blob/main/ansible/roles/schulcloud-server-core/templates/ingress.yml.j2#L7-L16
```suggestion
    nginx.ingress.kubernetes.io/ssl-redirect: "{{ TLS_ENABELD|default("false") }}"
    nginx.ingress.kubernetes.io/proxy-body-size: "{{ INGRESS_MAX_BODY_SIZE|default("2560") }}m"
    nginx.org/client-max-body-size: "{{ INGRESS_MAX_BODY_SIZE|default("2560") }}m"
    # The following properties added with BC-3606.
    # The header size of the request is too big. For e.g. state and the permanent growing jwt.
    # Nginx throws away the Location header, resulting in the 502 Bad Gateway.
    nginx.ingress.kubernetes.io/client-header-buffer-size: 100k
    nginx.ingress.kubernetes.io/http2-max-header-size: 96k
    nginx.ingress.kubernetes.io/large-client-header-buffers: 4 100k
    nginx.ingress.kubernetes.io/proxy-buffer-size: 96k
```

**Review by mamutmk5 — CHANGES_REQUESTED** [review]

File: ansible/roles/tldraw-client-core/templates/tldraw-client-ingress.yml.j2 (hunk 10 above)
Comment (mamutmk5) [reply ansible/roles/tldraw-client-core/templates/tldraw-client-ingress.yml.j2:9]: Okay, it's more then one line who must be removed

File: ansible/roles/tldraw-client-core/tasks/main.yml [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+  kubernetes.core.k8s:
+    kubeconfig: ~/.kube/config
+    namespace: "{{ NAMESPACE }}"
+    template: tldraw-client-deployment.yml.j2
+
+- name: TldrawClientIngress
+  kubernetes.core.k8s:
+    kubeconfig: ~/.kube/config
+    namespace: "{{ NAMESPACE }}"
+    template: tldraw-client-ingress.yml.j2
```
Comment (mamutmk5) [inline ansible/roles/tldraw-client-core/tasks/main.yml:24]: so let us **apply** the new ingress correctly.
Before it was an update with merge and the removed labels were still there. 
```suggestion
    template: tldraw-client-ingress.yml.j2
    apply: yes

```

File: ansible/roles/tldraw-client-core/templates/tldraw-client-deployment.yml.j2 [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+          name: cache-dir
+        - mountPath: /etc/nginx/conf.d
+          name: conf-dir
+        resources:
+          limits:
+            cpu: {{ TLDRAW_CLIENT_CPU_LIMITS|default("1000m", true) }}
+            memory: {{ TLDRAW_CLIENT_MEMORY_LIMITS|default("256Mi", true) }}
+          requests:
+            cpu: {{ TLDRAW_CLIENT_CPU_REQUESTS|default("100m", true) }}
+            memory: {{ TLDRAW_CLIENT_MEMORY_REQUESTS|default("32Mi", true) }}
```
Comment (mamutmk5) [inline ansible/roles/tldraw-client-core/templates/tldraw-client-deployment.yml.j2:69]: the memory is to even for dev.
```suggestion
            cpu: {{ TLDRAW_CLIENT_CPU_LIMITS|default("1000m", true) }}
            memory: {{ TLDRAW_CLIENT_MEMORY_LIMITS|default("2Gi", true) }}
          requests:
            cpu: {{ TLDRAW_CLIENT_CPU_REQUESTS|default("100m", true) }}
            memory: {{ TLDRAW_CLIENT_MEMORY_REQUESTS|default("256Mi", true) }}
```

**Review by mamutmk5 — CHANGES_REQUESTED** [review]

File: ansible/roles/tldraw-client-core/templates/tldraw-client-deployment.yml.j2 [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+  selector:
+    labels:
+      app: tldraw-client
+      app.kubernetes.io/part-of: schulcloud-verbund
+      app.kubernetes.io/version: {{ SCHULCLOUD_SERVER_IMAGE_TAG }}
+      app.kubernetes.io/name: tldraw-client
+      app.kubernetes.io/component: tldraw
+      app.kubernetes.io/managed-by: ansible
+      git.branch: {{ SCHULCLOUD_SERVER_BRANCH_NAME }}
+      git.repo: {{ SCHULCLOUD_SERVER_REPO_NAME }}
```
Comment (mamutmk5) [inline ansible/roles/tldraw-client-core/templates/tldraw-client-deployment.yml.j2:25]: sorry was in the wrong label line
```suggestion
  labels:
    app: tldraw-client
    app.kubernetes.io/part-of: schulcloud-verbund
    app.kubernetes.io/version: {{ SCHULCLOUD_SERVER_IMAGE_TAG }}
    app.kubernetes.io/name: tldraw-client
    app.kubernetes.io/component: tldraw
    app.kubernetes.io/managed-by: ansible
    git.branch: {{ SCHULCLOUD_SERVER_BRANCH_NAME }}
    git.repo: {{ SCHULCLOUD_SERVER_REPO_NAME }}
spec:
  replicas: {{ TLDRAW_CLIENT_REPLICAS|default("1", true) }}
  strategy:
    type: RollingUpdate
    rollingUpdate:
      maxSurge: 1
  revisionHistoryLimit: 4
  paused: false
  selector:
    labels:
      app: tldraw-client
```

**Review by mamutmk5 — CHANGES_REQUESTED** [review]

File: .github/workflows/clean.yml (hunk 1 above)
Comment (mamutmk5) [reply .github/workflows/clean.yml:10]: yes

File: .github/workflows/push.yml [hunk 15]
```diff
@@ -45,7 +45,7 @@ jobs:
 
       - name: Build and push ${{ github.repository }}
         if: ${{ env.IMAGE_EXISTS == 0 }}
-        uses: docker/build-push-action@v4
+        uses: docker/build-push-action@v4.1.1
```
Comment (mamutmk5) [inline .github/workflows/push.yml:48]: can go back to v4 with out an sub version

File: ansible/roles/tldraw-client-core/tasks/main.yml [hunk 16]
```diff
@@ -0,0 +1,25 @@
+- name: TldrawClientService
```
Comment (mamutmk5) [inline ansible/roles/tldraw-client-core/tasks/main.yml:1]: No Camel case, write it with spaces

File: ansible/roles/tldraw-client-core/tasks/main.yml [hunk 17]
```diff
@@ -0,0 +1,25 @@
...
+  kubernetes.core.k8s:
+    kubeconfig: ~/.kube/config
+    namespace: "{{ NAMESPACE }}"
+    template: tldraw-client-svc.yml.j2
+
+- name: TldrawClientConfigmap
```
Comment (mamutmk5) [inline ansible/roles/tldraw-client-core/tasks/main.yml:7]: No Camel case, write it with spaces

File: ansible/roles/tldraw-client-core/tasks/main.yml [hunk 18]
```diff
@@ -0,0 +1,25 @@
...
+  kubernetes.core.k8s:
+    kubeconfig: ~/.kube/config
+    namespace: "{{ NAMESPACE }}"
+    template: tldraw-client-configmap.yml.j2
+    apply: yes
+
+- name: TldrawClientDeployment
```
Comment (mamutmk5) [inline ansible/roles/tldraw-client-core/tasks/main.yml:14]: No Camel case, write it with spaces

File: ansible/roles/tldraw-client-core/tasks/main.yml [hunk 19]
```diff
@@ -0,0 +1,25 @@
...
+  kubernetes.core.k8s:
+    kubeconfig: ~/.kube/config
+    namespace: "{{ NAMESPACE }}"
+    template: tldraw-client-deployment.yml.j2
+
+- name: TldrawClientIngress
```
Comment (mamutmk5) [inline ansible/roles/tldraw-client-core/tasks/main.yml:20]: No Camel case, write it with spaces

**Review by mamutmk5 — CHANGES_REQUESTED** [review]

**Review by mamutmk5 — APPROVED** [review]

**Pull request merged**